bip-0174.mediawiki

```diff
@@ -305,6 +305,53 @@ Signers do not need to sign for all possible input types. For example, a signer
 
 A single entity is likely to be both a Signer and an Updater as it can update a PSBT with necessary information prior to signing it.
 
+====Data Signers Check For====
+
+For a Signer to only produce valid signatures for what it expects to sign, it must check that the following conditions are true:
+
+* If a non-witness UTXO is provided, it's hash must match the hash specified in the prevout
+* If a witness UTXO is provided, a witness signature is being produced
```

Comment:
Perhaps write this as "If a witness UTXO is provided, no non-witness signature may be created"? The current language sounds like a signature must be created.

Reply:
Done